lib/guardian/tag_guardian.rb

```diff
@@ -11,7 +11,8 @@ def can_tag_topics?
   end
 
   def can_tag_pms?
-    is_staff? && SiteSetting.tagging_enabled && SiteSetting.allow_staff_to_tag_pms
+    return false if @user.blank?
+    SiteSetting.tagging_enabled && @user == Discourse.system_user || (@user.group_users.map(&:group_id) & SiteSetting.pm_tags_allowed_for_groups.to_s.split("|").map(&:to_i)).any?
```

Comment:
```suggestion
    SiteSetting.tagging_enabled && @user == Discourse.system_user || @user.group_users.exists?(group_id: SiteSetting.pm_tags_allowed_for_groups.to_s.split("|").map(&:to_i))
```

Would this work instead so we do not have to initialise the `GroupUser` records.